components/analytics/format.tsx

```diff
@@ -99,14 +78,19 @@ export function formatUsersWithPostsPerDayData(data: any) {
             data: [] as any,
         }],
     };
+    return fillChartData(chartData, labels, data);
+}
 
-    for (const index in data) {
-        if (data[index]) {
-            const row = data[index];
-            chartData.labels.push(row.name);
-            chartData.datasets[0].data.push(row.value);
-        }
-    }
+function fillChartData(chartData: any, labels: any, data: any) {
+    if (data && data.length > 0) {
```

Comment:
```suggestion
    if (data?.length) {
``` 
Does this work ?

Reply:
added separately, because I ran into an issue before an wanted to test. But these suggestions all work...added.